src/diffusers/loaders.py

```diff
@@ -1074,6 +1086,28 @@ def lora_state_dict(
 
         return state_dict, network_alphas
 
+    @classmethod
+    def _best_guess_weight_name(cls, pretrained_model_name_or_path_or_dict, file_extension=".safetensors"):
+        targeted_files = []
+        if os.path.isdir(pretrained_model_name_or_path_or_dict):
+            targeted_files = [
+                f for f in os.listdir(pretrained_model_name_or_path_or_dict) if f.endswith(file_extension)
+            ]
+        else:
+            files_in_repo = model_info(pretrained_model_name_or_path_or_dict).siblings
+            targeted_files = [f.rfilename for f in files_in_repo if f.rfilename.endswith(file_extension)]
+
+        if len(targeted_files) == 0:
+            return
+
+        targeted_files = list(filter(lambda x: "scheduler" not in x and "optimizer" not in x, targeted_files))
+        if len(targeted_files) > 1:
+            raise ValueError(
```

Comment:
WDYT if this is a warning not a `raise`, and we pick the first one? Not a strong opinion tho.

Reply:
Better to raise an error than leaving it silent IMO. This promotes a sense to structuring the Hub repos too IMO.

Reply:
Fair. I thought about doing just a warning and still trying the best guess, like "we picked this, if you meant another one, improve your repo"

Reply:
Example use-case where this is useful: https://huggingface.co/Linaqruf/pastel-anime-xl-lora/tree/main

Reply:
Sure. But I think we need to meet at a common point and I would prefer the current design to promote what I said in the earlier comment.